node/src/main/java/de/obsidiancloud/node/module/ModuleLoader.java

```diff
@@ -0,0 +1,284 @@
+package de.obsidiancloud.node.module;
+
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** The module loader loads and unloads modules */
+public class ModuleLoader {
+    private static final List<ModuleManifest> manifests = new ArrayList<>();
+
+    /**
+     * Gets all loaded manifests
+     *
+     * @return All loaded manifests
+     */
+    public static @NotNull List<ModuleManifest> getManifests() {
+        return manifests;
+    }
+
+    /**
+     * Gets all loaded modules
+     *
+     * @return All loaded modules
+     */
+    @SuppressWarnings("unused")
+    public static @NotNull List<Module> getModules() {
+        ImmutableList.Builder<Module> modules = ImmutableList.builder();
+        for (ModuleManifest manifest : manifests) {
+            modules.add(manifest.getInstance());
+        }
+        return modules.build();
+    }
+
+    /**
+     * Gets a module by its name
+     *
+     * @param name The name of the module
+     * @return The module or null if the module is not loaded
+     */
+    public static @Nullable Module getModule(@NotNull String name) {
```

Comment:
Eventuell Modules, die enabled sind, in einer CopyOnWriteArrayList namens `enabledModules` o.Ä. speichern?

Reply:
juckt.